lib/backend/k8s/k8s_test.go

```diff
@@ -1769,3 +1769,65 @@ var _ = Describe("Test Syncer API for Kubernetes backend", func() {
 		})
 	})
 })
+
+var _ = Describe("Test Watch support", func() {
+	var (
+		c   *KubeClient
+		ctx context.Context
+	)
+
+	BeforeEach(func() {
+		// Create a client
+		cfg := apiconfig.CalicoAPIConfigSpec{KubeConfig: apiconfig.KubeConfig{K8sAPIEndpoint: "http://localhost:8080"}}
+		client, err := NewKubeClient(&cfg)
+		Expect(err).NotTo(HaveOccurred())
+		c = client.(*KubeClient)
+
+		ctx = context.Background()
+	})
+
+	It("Should support watching Nodes", func() {
+		By("Watching a single node", func() {
+			name := "127.0.0.1" // Node created by test/mock-node.yaml
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Name: name, Kind: apiv3.KindNode}, "")
+			Expect(err).NotTo(HaveOccurred())
+
+			// We should get at least one event from the watch.
+			var receivedEvent bool
+			for i := 0; i < 10; i++ {
```

Comment:
Should we make a change to the resource and see that we get that in the test, to make sure that the watch is setup correctly?

Reply:
I think this is enough to verify the watch is correct - we verify that we get at least one non-error event.